v1.0.0 public release

hansputera opened the pull request:
v1.0.0 bilibili-dl public release PR
---
**Todos**:
- [x] create `/api/playurl` endpoint to retrieve bilibili's video metadata (like video duration, video url, audio url, etc...).
- [x] create **video workers** with BullMQ to combine audio, and video together using FFMPeg.
- [ ] create `/api/download` to download video (explanation: see below)
- [ ] a simple site, and in the next version (probably v1.2.0), I will create a usable site (which is using FFMPeg-WASM).
- [x] create `__INITIAL_STATE__` extractor to get the anime metadata.
- [x] Video Meta Data Transformer
- [x] Fixing empty data result at `/api/playurl`, and `/api/search`.

---
## Explanation for `/api/download`
Basically, this endpoint is focused to download Bilibili's Video, and the response of this endpoint is job/queue id (uuid).

- Why the response is job/queue id?
Because, the video and audio are separate. And, it needs to be combined. So, we sent a job to worker to combine it.

> The plan for `/api/download` was canceled because we needed to host the workers. And, the host should be online for a long time (2-5 hours). I think, we can replace it with a website that can merge/merge them (audio, video).

**Links:**
1. The worker is available on https://github.com/hansputera/bilibili-dl-worker
2. To communicate with workers, we need workers controller https://github.com/hansputera/bilibili-dl-controller

hansputera committed:
feat(apps.web.pages.api.search): refetch when the result is an empty object

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(apps.web.pages.search.api): just save the results without additional changes

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(apps.web): added playurl endpoint

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(apps.web.pages.api): return "Couldn't get the video data" as a response

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.core): transform playurl data

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
fix(packages.util.transformers): fix missing audios, and videos prop

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util.transformers): just transform the duration, and size value when instanceToPlain

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
refactor(packages.core): return string instead as an Error than undefined

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
fix(apps.web.pages.api.playurl): return error message in object

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util): added window.__INITIAL_STATE__ extractor

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(apps.web.pages.api): added /api/meta endpoint

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.core): added getMeta function

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(apps.web.pages.api.meta): return extracted __INITIAL_STATE__ data-s

hansputera committed:
fix(packages.core): fix logic to detect unknown page

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(packages.core): change logic to detect unknown page

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
Merge branch 'dev' of https://github.com/hansputera/bilibili-dl into dev

hansputera committed:
refactor: change locale to en_US

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util.transformers): add /api/meta data transformer

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(packages.util): match season id, and video id

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(apps.web.pages.api.playurl): use video.videoId

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util.transformers.meta): add originTitle

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util): added matchView function

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(packages.util.transformers.search): use matchView function instead to get views count

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(packages.util.transformers.meta): added PhotoMeta, and MetaTransformed class

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(packages.util): change transform function name

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.core.meta): transform plain to instance

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(packages.core.playurl): use transformPlayUrl

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(apps.web.pages.api.meta): change response data

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
miss prop name

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util.transformers): add EpisodeMeta, and MetaTransformed.{episodes,limitAreas}

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util): added cleanupURL function

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util.transformers.meta): added clean episode url

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util.transformers.meta): added series

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util.transformers): add series on MetaTransformed

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore: added workspaces

hansputera committed:
chore: remove workspaces

hansputera committed:
fix(packages.core.search): fixed empty array results

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
fix(packages.util): fixed undefined "videoId" on getBtvID function

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
Merge branch 'dev' of https://github.com/hansputera/bilibili-dl into dev

Just git pull :-)

hansputera committed:
refactor(config): use api.bilibili.tv instead

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore(lockfile): update lockfile

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore(lockfile): update lockfile

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
chore(deps): remove bullmq

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.core): added timeout

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore: update lockfile

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.core.playurl): add some searchParams data

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.config.constants): added supportedLocales

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.core): added "locale" parameter to playurl, and search feature

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(apps.web.pages.api): implement locale

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor(apps.web.pages.api): use "enum" validation type

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
fix: fixed gateway timeout response, and empty object results

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
refactor: Implemented the new Bilibili interface.

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.core): added Origin, and Referer in request headers

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore(packages.config): added README

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(*): setup development stuff

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore: update lockfile

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore: add .vercelignore

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
fix(scripts.packs): restore the "main" property

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
I don't know what I'm doing

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore: use tsc instead

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
fix(packages.util.transformers.search): fix wrong video id

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore(packages.util.transformers.search): add todo

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.util): match anime url

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
fix(packages.core.playurl): remove "unknown error" match

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(packages.core.meta): use MetaTransformed as return type

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(apps.web.api.playurl): use first episode id if the given video id is season id

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(apps.web.api): create "filter" variable after parsing, and logical operation

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(apps.web.api): use "metaVideo.id" if the video isn't anime

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
feat(apps.web.api): create "v" variable in handler scope

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
fix(packages.util.transformers): fixed unknown sectionsList data

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore: update packages

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore(version): change package version

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore(packages.config): added constants.d.ts

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore(version.config): update to 1.0.1

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore: add packages.util#readme, and formatting all package.json

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
chore(packages.util#version): update to 1.0.4

Signed-off-by: GitHub <noreply@github.com>

hansputera committed:
chore(deps): update dependencies

Signed-off-by: Hanif Dwy Putra S <47862061+hansputera@users.noreply.github.com>

hansputera committed:
deps: update turbo

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
style: prettier format

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
fix(core.search): fix missing anime item

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
fix(util.meta): fix missing sectionsList on season data

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hanifdwyputras committed:
chore: migrate to yarn

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
chore: bump version

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
cleanup

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: use tsup

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
fix(packages.util): fix config/constants path

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: filter ugc, and ogv only

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
chore: cleanup

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
chore: update yarn

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
chore: styling

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
lint: fix lint jsdoc

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: add subtitle interfaces

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat(interfaces): add Subtitle data

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: add subtitle transformer

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
chore: remove test file

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: export subtitle

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
fix(subtitle): fix convertToSRTTime

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat(core): add subtitle

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
fix(core.subtitle): fix lang key

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
fix(*): fix stream end

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: add subtitle api #2

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
fix(core.subtitle): fix undefined subtitles

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
chore: bump version

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
chore: cleanup

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras commented:
https://github.com/hansputera/bilibili-dl/pull/1/commits/dd67c16944bb9fe8b0e16a7233843464f09cda22 reports:
- We need to fix the response.data filter (/api/search)

hanifdwyputras committed:
fix(core.search): ignore nav, and creator

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
style: formatting the json

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: just ignore the creator

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
chore: styling

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
fix(util.meta): fix sectionsList

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
fix(util.meta): use ogv.season OR ogv

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
style: format

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras commented:
> [dd67c16](https://github.com/hansputera/bilibili-dl/commit/dd67c16944bb9fe8b0e16a7233843464f09cda22) reports:
> 
>     * We need to fix the response.data filter (/api/search)

fixed https://github.com/hansputera/bilibili-dl/pull/1/commits/ad72ed3a97fa63e0256798773d7624012d60a589

hanifdwyputras committed:
feat: add middleware

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: add cors

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
feat: applying CORS

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hanifdwyputras committed:
styling

Signed-off-by: Hanif Dwy Putra S <99879343+hanifdwyputras@users.noreply.github.com>

hansputera committed:
fix(core): upgrade to v2 api

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
chore(core.search): remove import compare

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat(apps/web): add sentry

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
chore(deps): update windicss-webpack-plugin, and @sentry/nextjs

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
chore(lint): fix the lint rules

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
chore: remove useless word

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat(util): add between string extractors, and use it on initialState extractor

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat(apps/web): completing custom 404 page

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
fix(util.search): fix missing genres

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat(util.playurl)!: remove duration

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat(*.playurl): add frameRate, and backupUrls on Resource. And, remove unecessary queries

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat(scripts): add login bilibili

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat(scripts): add creds test script

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat(util.search): remove "Update at" word from update_pattern

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
fix(util.meta): fix empty publishDate

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
fix(util.meta): fix empty episodes

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

hansputera committed:
feat: add cookie

Signed-off-by: Hanif Dwy Putra S <hanifdwyputrasembiring@gmail.com>

Commit:
chore: upgrade nextjs version, migrate to unocss

Commit:
chore: uninstall deprecated package

Commit:
chore: migrate to tailwindcss

Commit:
chore: migrate to app directory

Commit:
fix: useRouter

Commit:
fix: eslint

chore: add automatic support nextjs types for typescript

hansputera committed:
Merge pull request #3 from Reyuki-san/migrate-next-13

Migrate to nextjs 13

realyukii committed:
Update! (#4)

* chore: update configuration for debugging and monitoring

* fix: unexpected cannot parse json, use text instead

* chore: add tailwind prettier plugin

* chore: add @material-tailwind/react

* chore: add @material-tailwind/react

* chore: add owner

* revert: return to pages dir

* chore: add JSDoc

* chore: add JSDoc

* feat: add layout, dark mode, path aliases

* fix: dark mode

feat: add logo

* feat: add following component

* feat: add i18n feature, update Header component

* fix: missing jsdoc

* feat: add switcher language

fix: disable usage of cookie as workaround for IntlError: MISSING_MESSAGE: No messages were configured on the provider.

chore: add semantic naming conventions for lang format

* feat: add display suggestion hot anime with mock data

* feat: Display popular anime with Custom Card

* fix: color theme

* feat: adjusts subtitle to json format

wip: responsive layout

* feat: add swr, carousel

wip: Anime timeline component

wip: get list of popular anime

* wip: get list of popular anime

* fix: eslint prefer-spread error

* feat: delete unused endpoint

* fix: vercel error - https://github.com/vercel/next.js/discussions/49269

* fix: vercel error https://github.com/vercel/next.js/issues/49355#issuecomment-1538406708

* wip: improve some responsive layout

feat: fill Recomendation, Popular, Anime timeline component with real content, then handle it with custom hooks

chore: add general interface

* fix: eslint
